os-fpga/FOEDAG
# [RG-240] Add Timing Analysis settings

> ### Motivate of the pull request
> - [x] To address an existing issue. If so, please provide a link to the issue: RG-240
> - [ ] Breaking new feature. If so, please describe details in the description part.

> ### Describe the technical details
> #### What is currently done? (Provide issue link if applicable)
> - Timing Analysis engine can only chosen at task execution with the tcl command `sta opensta` which will use OpenSta otherwise, `sta` defaults to tatum/OpenVpr
>
> #### What does this pull request change?
> - Timing Analysis now has task settings for the engine type:
>   ![image](https://user-images.githubusercontent.com/103447061/208496894-cba3bb69-b182-4c15-9413-e21405c321aa.png)
> - `TimingAnalysisEngineOpt()` has been added to Compiler.cpp
>    - This was necessary to maintain engine choice after a clean as the base TimingAnalysis option blends Clean operations in with engine selection which means when a user does a clean, their engine choice would be lost as well if we use TimingAnalysisOpt


> ### Which part of the code base require a change
> 
> - [x] Library: Compiler, Main
> - [ ] Plug-in: <Specify the plugin name>
> - [ ] Engine
> - [ ] Documentation
> - [ ] Regression tests
> - [ ] Continous Integration (CI) scripts

> ### Testing
> - Manually tested in Raptor:
>    - launched with `build/bin/raptor --script test/Testcases/oneff/raptor.tcl`
>    - Let run finish
>    - Double click clean under Timing Analysis
>    - Open Timing Anaylsis settings and change to OpenSTA
>    - Double Click Timing Analysis
>    - R-click Timing Analysis in task list and select `View Timing Analysis Logs`
>      - Ensure OpenSTA was used
>    - Double click clean under Timing Analysis
>    - Open Timing Anaylsis settings and change to Tatum
>    - Double Click Timing Analysis
>    - R-click Timing Analysis in task list and select `View Timing Analysis Logs`
>      - Ensure Tatum/OpenVpr was used
